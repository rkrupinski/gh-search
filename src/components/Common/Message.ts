import styled, { keyframes } from "styled-components";

const appear = keyframes`
  to { opacity: 1; }
`;

const icons: Record<MessageType, string> = {
  loading: "‚è≥",
  error: "üí©",
};

export type MessageType = "loading" | "error";

export type MessageProps = {
  type: MessageType;
  waitMs?: number;
};

export const Message = styled.p<MessageProps>`
  margin: calc(${({ theme }) => theme.layout.gutter} * 3) 0;
  text-align: center;
  word-wrap: break-word;
  color: ${({ theme }) => theme.colors.textSecondary};
  opacity: 0;
  animation-name: ${appear};
  animation-fill-mode: forwards;
  animation-duration: ${({ theme }) => theme.animation.duration};
  animation-delay: ${({ waitMs }) => `${waitMs ?? 0}ms`};

  &:before {
    content: ${({ type }) => `"${icons[type]}"`};
    margin-right: 0.25rem;
  }
`;
