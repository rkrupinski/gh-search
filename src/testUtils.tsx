import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { StoreProvider } from "store";
import { theme } from "theme";

const Providers: React.FC = ({ children }) => (
  <StoreProvider>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </StoreProvider>
);

export const renderWithProviders = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries" | "wrapper">,
) => render(ui, { wrapper: Providers, ...options });

export const noop = () => {
  /**/
};
