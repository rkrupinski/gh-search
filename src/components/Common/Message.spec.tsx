import React from "react";
import { Message } from "./Message";
import { renderWithProviders } from "testUtils";

describe("Message", () => {
  it("should render loading message", () => {
    const { container } = renderWithProviders(<Message type="loading">Loading</Message>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render error message", () => {
    const { container } = renderWithProviders(<Message type="error">Error</Message>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should let one delay the appearance", () => {
    const { container } = renderWithProviders(
      <Message type="loading" waitMs={1000}>
        Loading delayed
      </Message>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
