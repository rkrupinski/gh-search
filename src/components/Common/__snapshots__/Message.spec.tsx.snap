// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should let one delay the appearance 1`] = `
.c0 {
  margin: calc(1rem * 3) 0;
  text-align: center;
  word-wrap: break-word;
  color: #777;
  opacity: 0;
  -webkit-animation-name: efKvNv;
  animation-name: efKvNv;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-duration: 50ms;
  animation-duration: 50ms;
  -webkit-animation-delay: 1000ms;
  animation-delay: 1000ms;
}

.c0:before {
  content: "‚è≥";
  margin-right: 0.25rem;
}

<p
  class="c0"
  type="loading"
>
  Loading delayed
</p>
`;

exports[`Message should render error message 1`] = `
.c0 {
  margin: calc(1rem * 3) 0;
  text-align: center;
  word-wrap: break-word;
  color: #777;
  opacity: 0;
  -webkit-animation-name: efKvNv;
  animation-name: efKvNv;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-duration: 50ms;
  animation-duration: 50ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

.c0:before {
  content: "üí©";
  margin-right: 0.25rem;
}

<p
  class="c0"
  type="error"
>
  Error
</p>
`;

exports[`Message should render loading message 1`] = `
.c0 {
  margin: calc(1rem * 3) 0;
  text-align: center;
  word-wrap: break-word;
  color: #777;
  opacity: 0;
  -webkit-animation-name: efKvNv;
  animation-name: efKvNv;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-duration: 50ms;
  animation-duration: 50ms;
  -webkit-animation-delay: 0ms;
  animation-delay: 0ms;
}

.c0:before {
  content: "‚è≥";
  margin-right: 0.25rem;
}

<p
  class="c0"
  type="loading"
>
  Loading
</p>
`;
